internal/pkg/agent/cmd/status.go

```diff
@@ -145,6 +147,43 @@ func listAgentState(l list.Writer, state *client.AgentState, all bool) {
 	}
 	l.UnIndent()
 	listComponentState(l, state.Components, all)
+
+	// Upgrade details
+	listUpgradeDetails(l, state.UpgradeDetails)
+}
+
+func listUpgradeDetails(l list.Writer, upgradeDetails *cproto.UpgradeDetails) {
+	if upgradeDetails == nil {
+		return
+	}
+
+	l.AppendItem("upgrade_details")
+	l.Indent()
+	l.AppendItem("target_version: " + upgradeDetails.TargetVersion)
+	l.AppendItem("state: " + upgradeDetails.State)
+	if upgradeDetails.ActionId != "" {
+		l.AppendItem("action_id: " + upgradeDetails.ActionId)
+	}
+
+	if upgradeDetails.Metadata != nil {
```

Comment:
https://pkg.go.dev/google.golang.org/protobuf/encoding/prototext might be a way to avoid having to keep this up to date over time, but the output might not look as nice.